magma/bits.py

```diff
@@ -412,7 +417,10 @@ def __and__(self, other):
             return NotImplemented
 
     def __rand__(self, other):
-        return self & other
+        try:
```

Comment:
would consider using decorator like

```python
def _not_implemented_on_type_error(fn):

    @functools.wraps(fn)
    def _wrapper(*args, **kwargs):
        try:
            return fn(*args, **kwargs)
        except TypeError:
            return NotImplemented

    return _wrapper
```

Actually, could make it even more general if wanted (and put in `common.py`):

```python
def return_on_exception(func=None, *, exceptions=(Exception,), return_value=None):
    if func is None:
        return functools.partial(return_on_exception, exceptions=exceptions, return_value=return_value)

    @functools.wraps(func)
    def _wrapper(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except exceptions:
            return return_value

    return _wrapper
```

This could then be used like

```python
@return_on_exception((TypeError,), NotImplemented)
def __rand__(self, other):
    ...
```

(could also do like `roperator_decorator = return_on_exception((TypeError,), NotImplemented)` for shorthand on each op)

Reply:
I think there was some reason we couldn't use a decorator (this is the pattern that hwtypes uses as well), but let me see if I can remember

Reply:
Hmm couldn't seem to remember/find any comments on the old related PRs.

I used the first decorator approach since I also refactored the
```
        except InconsistentSizeError as e:
            raise e from None
```
and this wasn't quite as easy to capture as function arguments (couldn't put it in a lambda)